cips/cip-29.md

```diff
@@ -0,0 +1,199 @@
+
+At a 33% reduction the inflation rate is 4.82% in the first year of the reduction, and the APR is roughly 7.39% if the bonding ratio stays the same. This is currently around 2.2 times the APR of Ethereum staking, but Celestia is a smaller and younger network so a higher security budget is justified.
+
+## **Specification**
+
+### **Overview**
+
+1. Reduce Inflation: We reduce the current inflation by 33% in the next upgrade (v4)  
+2. Reduce Disinflation Rate: We reduce the disinflation rate by 33%  
+3. Reward Calculation:
+
+   * The reward distribution mechanism remains unchanged—rewards are still allocated pro rata to stakers and validators.
+
+   * The only modification is to the base inflation parameter and the disinflation parameter; both drop by 33% on the next upgrade leading to lower yet sustainable inflation immediately
+
+Implementers MUST ensure:
+
+* The chain upgrade process includes the new inflation parameters without disrupting block production.
+
+* The new schedule is included in the next major version release to reflect the updated inflation rates on-chain in the next app version (v4).
+
+## **Parameters**
+
+Below is the illustrative table showing the original versus the accelerated schedule. The “New Adjusted Inflation” column is an example of how inflation might look if we reduce it by 33% starting in the next upgrade. Note that for simplicity, the table assumes the inflation drop is applied at year 1.5 but if this proposal gets accepted, it would be immediately applied with the next upgrade.
+
+| Year    | Original Inflation | New Adjusted Inflation | Notes                                                                                                                        |
+|:--------|:-------------------|:-----------------------|:-----------------------------------------------------------------------------------------------------------------------------|
+| **0**   | 8.00               | 8.00                   | Genesis year, no change.                                                                                                     |
+| **1**   | 7.20               | 7.20                   | First disinflation applied (10%).                                                                                            |
+| **1.5** | 7.20               | 4.82                   | 33% drop in inflation applied, disinflation rate reduced to 6.7%. (This will be implemented in the next upgrade of Celestia) |
+| **2**   | 6.48               | 4.50                   | Regular annual disinflation applied (6.7%).                                                                                  |
```

Comment:
I'm calculating different numbers for the adjusted inflation. Here are the existing constants:

```
initialInflation = .08
disinflation = .1
```

The "New Adjusted Inflation" column is decreasing both of those by 33% so

```
initialInflation = .0536
disinflation = .067
```

The formula to calculate inflation rate in a given year is: 

```
InitialInflationRate * ((1 - DisinflationRate) ^ YearsSinceGenesis)
```

let's skip year 0 and 1 because we can hard-code them to use the previous constants. For year 1.5

```
.0536 * (1 - .067) ^ 1 = 0.0500088
```

For year 2

```
.0536 * (1 - .067) ^ 2 = 0.04665821
```

which diverge from the values in the table.

Reply:
I think it's the formula that might not be correct. I just double checked the table in a spreadsheet.

Reply:
Yea my formula is incorrect. 

```
InitialInflationRate * ((1 - DisinflationRate) ^ YearsSinceGenesis)
```

 no longer works because this proposal modifies the inflation rate and disinflation rate for years > 1.5

Reply:
Even if I use the reference implementation in this CIP, I get different numbers from the ones in the table. See https://github.com/celestiaorg/celestia-app/pull/4299